src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -430,6 +441,23 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
                 new BufferedReader(new InputStreamReader(partitionInputStream)),
                 CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
         iterator = parser.iterator();
+
+        try {
+          if (!iterator.hasNext()) {
+            flagAsIncomplete = true;
+          }
+        } catch (Exception e) {
+          /*
+           * FIX ME: If we hit an exception for a malformed row (in case of quotes for example), we mark this as the
+           * last row
+           */
+          if (e.getMessage() != null && e.getMessage().contains(truncatedQuoteErrorMessage)) {
+            onLastRow = true;
+          } else {
+            throw e;
+          }
+        }
+
         for (int i = 0; i < numberOfSkippedLines; i++) {
```

Comment:
Yes, this should be conditioned for the first worker....

But headers in subsequent partitions? How likely would that happen given that we try to keep the size of the partitions at a minimum number (believe it was 100MB). I guess a case where there is a very long header?